scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# can be run in 2 modes:
+# 1- $0 --hash $oldhash $newhash              - replace all instances of $oldhash with $newhash
+# 2- $0 --image $image $newhash        - replace all hashes for $image with $newhash
+#
+# for backwards compatibility if no argtype is given, the first method is assumed. Thus the following are equivalent:
+#   $0 $oldhash $newhash 
+#   $0 --hash $oldhash $newhash 
+#
+
+function usage() {
+    echo "Modes of operation:" >&2
+    echo >&2
+    echo "Replace by hash:" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo "	Example: $0 8675309abcdefg abcdef567899" >&2
```

Comment:
I am confused. I thought the two modes were:

1. Existing: replace all instances of hash x with hash y
2. New: change tags for all instances of image q with hash y

Are you asking that we change the first one to: replace all instances of image:tag combo q:x with q:y? I would think the existing functionality of hash is enough?